src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -110,7 +156,7 @@ private void initialize(String sasKeyName, String sasKey, String host, ServiceBu
         
         String serviceBusQuery = settings.getServiceBusQuery();
         destinationUri += serviceBusQuery;
-        this.factory = new JmsConnectionFactory(sasKeyName, sasKey, destinationUri);
+        this.factory = new JmsConnectionFactory(keyName, keyValue, destinationUri);
```

Comment:
> this.factory = new JmsConnectionFactory(keyName, keyValue, destinationUri); [](http://example.com/codeflow?start=7&length=76)

Can we add a function like GetJMSConnectionFactory and consolidate the getting of factory b/n when token is specified and when we are using keyName and keyValue. It'll make it easier to read.

Also in the current mechanism, we define a factory and then override it.

Also, done this way, when we use the factory generated under isToken block, the factory will not have the AMQP_OPEN_PROPERTIES applied to it.

Reply:
That was to address subscriptions but since we add fix in the backend everything will revert back as the original only add a separation for the jms password override and that will use the isAadAuthenticatioin.